src/esphomelib/sensor/tx20_component.cpp

```diff
@@ -0,0 +1,160 @@
+// Based on:
+
+#include "esphomelib/defines.h"
+
+#ifdef USE_TX20
+
+#include "esphomelib/espmath.h"
+#include "esphomelib/log.h"
+#include "esphomelib/text_sensor/text_sensor.h"
+#include "tx20_component.h"
+
+#ifdef ARDUINO_ARCH_ESP32
+  #define GPIO_STATUS_W1TC_ADDRESS 0x24
+#endif
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace sensor {
+
+static const char *TAG = "sensor.tx20";
+
+const char UNIT_KMH[] = "km/h";
+const char ICON_WIND_SPEED[] = "mdi:weather-windy";
+const char ICON_WIND_DIRECTION[] = "mdi:sign-direction";
+const char EMPTY[] = "";
+const uint16_t TX20_BIT_TIME = 1220;   // microseconds
+const uint8_t TX20_RESET_VALUES = 60;  // seconds
+
+const std::vector<std::string> DIRECTIONS = {"N", "NNE", "NE", "ENE", "E", "ESE", "SE", "SSE",
+                                             "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW"};
+
+static bool tx20_available = false;
+static float tx20_wind_speed_kmh = 0;
+static uint8_t tx20_wind_direction = 0;
```

Comment:
Don't make these global variables. Even though only one of these can exist at a time, please make them class members.